From 11703001b3e9c4c4af87e838a8d30f523fdf9b39 Mon Sep 17 00:00:00 2001
From: Christian Seiler <christian@iwakd.de>
Date: Sun, 4 Oct 2015 13:11:06 +0200
Subject: [PATCH 2/2] adt-virt-qemu: emulate host-type CPU

In order to make nested KVM work, the emulated CPU needs to support
virtualization. The easiest way to do that that doesn't require lots of
compatibility checks is to tell QEMU to emulate the host CPU. This will
only work in KVM mode, so only enable it there.

This does not guarantee that nested KVM will work (the host's KVM
module has to have the appropriate option set), but it is required.
---
 virt-subproc/adt-virt-qemu | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/virt-subproc/adt-virt-qemu b/virt-subproc/adt-virt-qemu
index 2bed237..85e7bbd 100755
--- a/virt-subproc/adt-virt-qemu
+++ b/virt-subproc/adt-virt-qemu
@@ -495,6 +495,10 @@ def hook_open():
 
     if os.path.exists('/dev/kvm'):
         argv.append('-enable-kvm')
+        # emulate host CPU so that nested KVM might work (if it's
+        # enabled)
+        argv.append('-cpu')
+        argv.append('host')
 
     # pass through option to qemu
     if args.qemu_options:
-- 
2.1.4

